hls4ml/backends/quartus/quartus_backend.py

```diff
@@ -188,3 +201,56 @@ def init_softmax(self, layer):
         if 'inv_table_t' not in layer.attributes:
             layer.set_attr('inv_table_t', layer.get_attr('table_t'))
+
+    @layer_optimizer(LSTM)
+    def init_lstm(self, layer):
+        # TODO Allow getting recurrent reuse factor from the config
+        reuse_factor = layer.model.config.get_reuse_factor(layer)
+        layer.set_attr('recurrent_reuse_factor', reuse_factor)
+
+        recurrent_bias = np.zeros(layer.weights['recurrent_weight'].shape[1])
+        layer.add_weights_variable(name='recurrent_bias', var_name='br_{index}', data=recurrent_bias)
+
+        recurrent_bias_i = np.zeros(layer.weights['recurrent_weight_i'].shape[1])
+        layer.add_weights_variable(name='recurrent_bias_i', var_name='br_i_{index}', data=recurrent_bias_i)
+
+        recurrent_bias_f = np.zeros(layer.weights['recurrent_weight_f'].shape[1])
+        layer.add_weights_variable(name='recurrent_bias_f', var_name='br_f_{index}', data=recurrent_bias_f)
+
+        recurrent_bias_c = np.zeros(layer.weights['recurrent_weight_c'].shape[1])
+        layer.add_weights_variable(name='recurrent_bias_c', var_name='br_c_{index}', data=recurrent_bias_c)
+
+        recurrent_bias_o = np.zeros(layer.weights['recurrent_weight_o'].shape[1])
+        layer.add_weights_variable(name='recurrent_bias_o', var_name='br_o_{index}', data=recurrent_bias_o)
+
+        index_t = IntegerPrecisionType(width=1, signed=False)
+
+        if 'table_t' not in layer.attributes:
+            layer.set_attr('table_t', FixedPrecisionType(width=18, integer=8))
+        if 'table_size' not in layer.attributes:
+            layer.set_attr('table_size', 1024)
+        if layer.model.config.is_resource_strategy(layer):
+            n_in, n_out, n_in_recr, n_out_recr = self.get_layer_mult_size(layer)
+            self.set_closest_reuse_factor(layer, n_in, n_out)
+            self.set_closest_reuse_factor(layer, n_in_recr, n_out_recr, attribute='recurrent_reuse_factor')
+
+            layer.weights['weight'].data = np.transpose(layer.weights['weight'].data)
+            layer.weights['recurrent_weight'].data = np.transpose(layer.weights['recurrent_weight'].data)
+
+            layer.weights['weight_i'].data = np.transpose(layer.weights['weight_i'].data)
+            layer.weights['recurrent_weight_i'].data = np.transpose(layer.weights['recurrent_weight_i'].data)
+
+            layer.weights['weight_f'].data = np.transpose(layer.weights['weight_f'].data)
+            layer.weights['recurrent_weight_f'].data = np.transpose(layer.weights['recurrent_weight_f'].data)
+
+            layer.weights['weight_c'].data = np.transpose(layer.weights['weight_c'].data)
+            layer.weights['recurrent_weight_c'].data = np.transpose(layer.weights['recurrent_weight_c'].data)
+
+            layer.weights['weight_o'].data = np.transpose(layer.weights['weight_o'].data)
+            layer.weights['recurrent_weight_o'].data = np.transpose(layer.weights['recurrent_weight_o'].data)
```

Comment:
Transposing could be moved to a resource strategy optimizer, as is done for many layers in Vivado. I added the basic optimizer in this commit: https://github.com/fastmachinelearning/hls4ml/pull/561/commits/068dcc7086cd966625006726b04a5689537cc587 for Conv layers (still unmerged)